src/controller/python/chip/yaml/parser.py

```diff
@@ -29,11 +29,170 @@
+        self._type = constraints.get('type')
+        self._starts_with = constraints.get('startsWith')
+        self._ends_with = constraints.get('endsWith')
+        self._is_upper_case = constraints.get('isUpperCase')
+        self._is_lower_case = constraints.get('isLowerCase')
+        self._min_value = _ConstraintValue(constraints.get('minValue'), field_type,
+                                           reference_objects)
+        self._max_value = _ConstraintValue(constraints.get('maxValue'), field_type,
+                                           reference_objects)
+        self._contains = constraints.get('contains')
+        self._excludes = constraints.get('excludes')
+        self._has_masks_set = constraints.get('hasMasksSet')
+        self._has_masks_clear = constraints.get('hasMasksClear')
+        self._not_value = _ConstraintValue(constraints.get('notValue'), field_type,
+                                           reference_objects)
+
+    def are_constrains_met(self, response) -> bool:
+        return_value = True
+
+        if self._has_value:
+            logger.warn(f'HasValue constraint currently not implemented, forcing failure')
+            return_value = False
+
+        if self._type:
+            logger.warn(f'Type constraint currently not implemented, forcing failure')
+            return_value = False
+
+        if self._starts_with and not response.startswith(self._starts_with):
+            return_value = False
+
+        if self._ends_with and not response.endswith(self._ends_with):
+            return_value = False
+
+        if self._is_upper_case and not response.isupper():
+            return_value = False
+
+        if self._is_lower_case and not response.islower():
+            return_value = False
+
+        min_value = self._min_value.get_value()
+        if response is not NullValue and min_value and response < min_value:
+            return_value = False
+
+        max_value = self._max_value.get_value()
+        if response is not NullValue and max_value and response > max_value:
+            return_value = False
+
+        if self._contains and not set(self._contains).issubset(response):
+            return_value = False
+
+        if self._excludes and not set(self._excludes).isdisjoint(response):
+            return_value = False
+
+        if self._has_masks_set:
+            for mask in self._has_masks_set:
+                if not (mask & response):
+                    return_value = False
+
+        if self._has_masks_clear:
+            for mask in self._has_masks_clear:
+                if mask & response:
+                    return_value = False
+
+        not_value = self._not_value.get_value()
+        if not_value and response == not_value:
+            return_value = False
+
+        return return_value
+
+
+class _SaveAs:
+    def __init__(self, save_as_key: str, response_storage: ResponseStorage):
+        self._save_as_key = save_as_key
```

Comment:
Suggest rename of "save_as_key" to "variable_name". Let's call them variables, rather than keys